### DISCUSSION TITLE
TDS step size confusion

### DISCUSSION THREAD

**Discussion by TheASingh** [CURENT/andes#266] [Q&A] [answered]
Hi Dr. Cui,

I ran the TDS for **1s**, after exporting the result file, I found that there were **102 records** in the output csv file.

I also tried other values for time frame and following are the outcomes:
**time frame 2s, 103 records**
**time frame 5s, 154 records**
**time frame 10s, 305 records**
**time frame 20s, 607 records**

I used default `tstep `(1/30). 
As I understand, there should have been 30 records for TDS (of 1s), 60 records for TDS (of 2s) and so on.

Please clear my doubts.

Regards,
A Singh

**Comment by cuihantao** [comment, marked as answer]
The time step size will be adjusted for convergence. You can export the data file to csv and inspect the timestamps.

`andes plot output.ls --to-csv`

**Reply by cuihantao** [reply]
There is one config flag, `shrinkt`, in `[TDS]` for controlling if the step size should be shinked. It is set to 1 by default.

If you change it to `0`, the total timestamps should be 30 * t_simulate +1 (for the initial) plus a few extra steps around the fault times.

**Reply by TheASingh** [reply]
Thanks, it worked for me.
